From e1b135167e027632564765a2ee7c8ceb473704cb Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 14 Feb 2023 18:02:04 +0200
Subject: [PATCH] [SW-66329] pif_arc: stop timer before rearm

pif timer is being rearmed before each time FLR is being handled,
because it is used to monitor the FLR timeout condition according to
the PciE spec.
In order to reset the timer it must be stopped first, and then the
timstamp register value can be overridden.

Change-Id: I566c9563c7cbdf9fc17a76fcb1ea3ca1094cc1c8
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/lib/common/pif_ts.c b/lib/common/pif_ts.c
index df040ff..6ca8238 100644
--- a/lib/common/pif_ts.c
+++ b/lib/common/pif_ts.c
@@ -15,6 +15,7 @@
 
 void pif_ts_counter_arm(void)
 {
+	regmap_writel(0, pif_arc.timestamp_ctrl);
 	regmap_writel(0, pif_arc.timestamp_cnt);
 	regmap_writel(PIF_TS_CNT_ENABLE, pif_arc.timestamp_ctrl);
 }
